fuse_optimizers/include/fuse_optimizers/fixed_size_smoother.h

```diff
@@ -0,0 +1,197 @@
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_OPTIMIZERS_FIXED_SIZE_SMOOTHER_H
+#define FUSE_OPTIMIZERS_FIXED_SIZE_SMOOTHER_H
+
+#include <fuse_core/graph.h>
+#include <fuse_core/transaction.h>
+#include <fuse_core/uuid.h>
+#include <fuse_optimizers/fixed_size_smoother_params.h>
+#include <fuse_optimizers/variable_stamp_index.h>
+#include <fuse_optimizers/windowed_optimizer.h>
+#include <ros/node_handle.h>
+#include <ros/time.h>
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace fuse_optimizers
+{
+/**
+ * @brief A fixed-Size smoother implementation that marginalizes out variables that are older than a defined Size time
+ *
+ * This implementation assumes that all added variable types are either derived from the fuse_variables::Stamped class,
+ * or are directly connected to at least one fuse_variables::Stamped variable via a constraint. The current time of
+ * the fixed-Size smoother is determined by the newest stamp of all added fuse_variables::Stamped variables.
+ *
+ * During optimization:
+ *  (1) new variables and constraints are added to the graph
+ *  (2) the augmented graph is optimized and the variable values are updated
+ *  (3) all motion models, sensors, and publishers are notified of the updated graph
+ *  (4) all variables older than "current time - Size duration" are marginalized out.
+ *
+ * Optimization is performed at a fixed frequency, controlled by the \p optimization_frequency parameter. Received
+ * sensor transactions are queued while the optimization is processing, then applied to the graph at the start of the
+ * next optimization cycle. If the previous optimization is not yet complete when the optimization period elapses,
+ * then a warning will be logged but a new optimization will *not* be started. The previous optimization will run to
+ * completion, and the next optimization will not begin until the next scheduled optimization period.
+ *
+ * Parameters:
+ *  - Size_duration (float, default: 5.0) The duration of the smoothing window in seconds
```

Comment:
```suggestion
 *  - num_states (int, default: 10) The maximum number of states in the smoothing window
```
Or something similar